src/api/Base.js

```diff
@@ -341,9 +341,8 @@ class Base {
 
         // $FlowFixMe
         const xhrMethod: Function = this.xhr[method].bind(this.xhr);
-
         try {
-            const { data } = await xhrMethod({ id: getTypedFileId(id), url, ...requestData });
+            const { data } = await xhrMethod({ ...requestData, id: getTypedFileId(id), url });
```

Comment:
Not sure if this should be done. And likely will break stuff. Because now you are sending a fileId always and in some cases we want a folder id. Don't think any id should be sent here, and callers should be sending those

Reply:
Base.js should not have anything related to file or folder

Reply:
I'll move it back

The reason why we set the ID here is to help enforce that the ID is getting typed correctly. If folder related stuff wants to start using this then we will need to move this out.